components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -699,6 +636,95 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
         }
     }
 
+    private List<BasicOrganization> getOrganizationsList(boolean allSubOrgs, boolean recursive, Integer limit,
+                                                         String organizationId, String sortOrder,
+                                                         List<ExpressionNode> expressionNodes,
+                                                         List<ExpressionNode> parentIdExpressionNodes)
+            throws OrganizationManagementServerException {
+
+        FilterQueryBuilder filterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQuery(expressionNodes, filterQueryBuilder);
+        Map<String, String> filterAttributeValue = filterQueryBuilder.getFilterAttributeValue();
+        List<String> timestampTypeAttributes = filterQueryBuilder.getTimestampFilterAttributes();
+
+        FilterQueryBuilder parentIdFilterQueryBuilder = new FilterQueryBuilder();
+        appendFilterQueryForParentId(parentIdFilterQueryBuilder, parentIdExpressionNodes);
+        Map<String, String> parentIdFilterAttributeValueMap = parentIdFilterQueryBuilder.getFilterAttributeValue();
+        String parentIdFilterQuery = parentIdFilterQueryBuilder.getFilterQuery();
+
+        String sqlStmt;
+        String getOrgSqlStmtTail = GET_ORGANIZATIONS_TAIL;
```

Comment:
Can we use this query itself if `allSubOrgs = true` and the DB is MySQL?

Reply:
I missed this DB, will check on this